ClangSharp.PInvokeGenerator.Test/PInvokeGeneratorTest.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace ClangSharp.Test
+{
+    public abstract class PInvokeGeneratorTest
+    {
+        protected const string DefaultLibraryPath = "ClangSharpPInvokeGenerator";
+        protected const string DefaultNamespaceName = "ClangSharp.Test";
+
+        protected const CXTranslationUnit_Flags DefaultTranslationUnitFlags = CXTranslationUnit_Flags.CXTranslationUnit_SkipFunctionBodies          // Don't traverse function bodies
+                                                                            | CXTranslationUnit_Flags.CXTranslationUnit_IncludeAttributedTypes      // Include attributed types in CXType
+                                                                            | CXTranslationUnit_Flags.CXTranslationUnit_VisitImplicitAttributes;    // Implicit attributes should be visited
+
+        protected static readonly string[] DefaultClangCommandLineArgs = new string[]
+        {
+            "-xc++",                                // The input files are C++
+            "-Wno-pragma-once-outside-header"       // We are processing files which may be header files
+        };
+
+        protected async Task ValidateGeneratedBindings(string inputContents, string expectedOutputContents)
+        {
+            // Ideally we would also have the input file be a MemoryStream. Unfortunately,
+            // Clang currently requires that all files exist on disk before Parse is called.
```

Comment:
Everything is setup to enable this if Clang also supported it, but it turns out `CXUnsavedFile` requires `CXUnsavedFile.Filename` to already exit on disk.

Perhaps we could "make it work" by having an empty file with a well-known name as part of the build output for the test library...

Reply:
This did work, and I've pushed a commit so that everything is now "in memory".